contracts/reserve/tobin_tax.go

```diff
@@ -0,0 +1,61 @@
+package reserve
+
+import (
+	"encoding/binary"
+	"errors"
+	"math/big"
+
+	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/contracts"
+	"github.com/celo-org/celo-blockchain/core/vm"
+	"github.com/celo-org/celo-blockchain/params"
+)
+
+var (
+	ErrTobinTaxZeroDenominator  = errors.New("tobin tax denominator equal to zero")
+	ErrTobinTaxInvalidNumerator = errors.New("tobin tax numerator greater than denominator")
+)
+
+type Ratio struct {
+	Numerator, Denominator *big.Int
+}
+
+func (r *Ratio) Apply(value *big.Int) *big.Int {
```

Comment:
Is this used anywhere?

Reply:
Wait, see it used below, but this feels like a weird thing to export.

Reply:
why weird to export?

TobinTax returns a Ratio, so if TobinTax is public, so is Ratio... and so is `Apply`. idea here to avoid other need to do the math themselves.